src/SFA.DAS.AssessorService.Data/RegisterRepository.cs

```diff
@@ -159,18 +159,43 @@ public async Task<string> CreateEpaOrganisationStandard(EpaOrganisationStandard
                 "UPDATE [OrganisationStandard] SET [DateStandardApprovedOnRegister] = getutcdate() where Id = @osdaId and [DateStandardApprovedOnRegister] is null",
                     new { osdaId });
 
-            await _unitOfWork.Connection.ExecuteAsync(
-                @"UPDATE [OrganisationStandardVersion] 
-                        SET [EffectiveFrom] = @effectiveFrom,
-                            [EffectiveTo] = @effectiveTo
-                        WHERE
-                            [OrganisationStandardId] = @id",
-                new
+
+            if (null != orgStandard.StandardVersions && applyFollowingWithdrawal)
+            {
+                foreach (var version in orgStandard.StandardVersions)
                 {
-                    orgStandard.EffectiveFrom,
-                    orgStandard.EffectiveTo,
-                    orgStandard.Id
-                });
+                    var standardUid = $"{orgStandard.StandardReference.Trim()}_{version.Trim()}";
+
+                    await _unitOfWork.Connection.ExecuteAsync(
```

Comment:
Can this be tweaked a bit? 
The code below detects for each individual org standard version, and then inserts if it doesn't exist, but while you're doing the insert, you could insert the correct effective from and effective to meaning you wouldn't need to update it afterwards?

You almost want to query for all versions of the OrgStandard First, then based on what exists and what doesn't, then update / create.

Alternatively, can you change the SQL so the IF NOT EXISTS has an else?
- IF NOT EXISTS
- INSERT INTO
- ELSE UPDATE

What do you prefer?

Reply:
Agreed - added IF ELSE for SQL Query

```
IF NOT EXISTS
    INSERT
ELSE
   UPDATE
```